pg_yregress/instance.c

```diff
@@ -204,14 +204,25 @@ yinstance_connect_result yinstance_connect(yinstance *instance) {
   return yinstance_connect_failure;
 }
 
-void yinstance_start(yinstance *instance) {
+void yinstance_prepare(yinstance *instance) {
   if (instance->managed) {
+    char datadir_template[] = "pg_yregress_XXXXXX";
+    char *datadir = mkdtemp(datadir_template);
+    if (datadir == NULL) {
+      fprintf(stderr, "Failed to create temporary directory: %s\n", strerror(errno));
```

Comment:
TODO: should be a boolean function to indicate whether this succeeded or we could bail out of the whole process as well